src/lib/components/tree/node-tree/terra-node-tree.component.ts

```diff
@@ -151,7 +151,7 @@ export class TerraNodeTreeComponent<D> implements OnDestroy, OnInit
         }
 
         // search node names if no tags found
-        if(!hasValidCaptionOrTag)
+        if(!hasValidCaptionOrTag && !isNullOrUndefined(node.name))
```

Comment:
Ich muss gestehen, ich verstehe diese ganze Methode nicht so recht :-D 
Was macht die?

Reply:
ich glaube das naming is bekackt XD dort wird nach tags oder namen gesucht, die in der suche eingegeben werden